### ISSUE TITLE
Evaluate datasets without train_set

### ISSUE THREAD

**Issue by xin-jin** [Microsoft/LightGBM#1937]
I loaded a Booster I previously trained, and tried to evaluate its performance on some new dataset. However eval() seems to require the `train_set` attribute of Booster. Is it possible to eval without it? 

Thanks!

**Comment by guolinke** [comment]
you can call `predict` and use the predicted result in the eval function written by yourself.

**Comment by xin-jin** [comment]
> you can call `predict` and use the predicted result in the eval function written by yourself.

Thanks for the reply. That's exactly what I have been doing. However, one strange thing is that the validation set result (binary logloss by accessing `model.best_score['valid_0']['binary_logloss']`) is always different from what sklearn logloss function gives from the predictions (not a huge amount, but obvious difference; sorry I don't have the numbers at hand right now). I also implemented my own version of logloss, and the result is the same as sklearn. Could you think of any possible reasons for the difference? I've checked that the model is using the same iteration for both validation and predictions.

**Comment by guolinke** [comment]
I think the slight difference is normal, as there are some precision problems for floating computation. 
The logloss in lightgbm is calculated by cpp, therefore, it may is different with the result calculated by python.
